sound/soc/sof/intel/hda-codec.c

```diff
@@ -29,10 +29,15 @@ static void hda_codec_load_module(struct hda_codec *codec)
 {
 	char alias[MODULE_NAME_LEN];
 	const char *module = alias;
+	int res;
 
 	snd_hdac_codec_modalias(&codec->core, alias, sizeof(alias));
 	dev_dbg(&codec->core.dev, "loading codec module: %s\n", module);
 	request_module(module);
+
+	res = device_attach(hda_codec_dev(codec));
+	if (res < 0)
+		dev_err(&codec->core.dev, "codec driver attach fail %d\n", res);
```

Comment:
if this can fail, we should return the error no?

Reply:
@ranj063 Probably so yes. We do catch the error in the ASoC card registration. I want to also study the device_*() API a bit more, but I wanted to get a version of the patch uploaded so this can be tested on target hardware ASAP.